common/odigos_config.go

```diff
@@ -1,5 +1,241 @@
 package common
 
+import (
+	"fmt"
+	"sort"
+
+	"github.com/odigos-io/odigos/common/consts"
+)
+
+type ConfigField interface {
+	ToString()
+}
+
+type ConfigBool bool
+
+func (data ConfigBool) ToString() {
+	fmt.Printf(": %t\n", data)
+}
+
+type ConfigBoolPointer struct {
+	Value *bool
+}
+
+func (data ConfigBoolPointer) ToString() {
+	if data.Value == nil {
+		fmt.Printf(": not set\n")
+	} else {
+		fmt.Printf(": %v\n", *data.Value)
+	}
+}
+
+type ConfigInt int
+
+func (data ConfigInt) ToString() {
+	fmt.Printf(": %d\n", data)
+}
+
+type ConfigString string
+
+func (data ConfigString) ToString() {
+	if data == "" {
+		fmt.Println(": not set")
+	} else {
+		fmt.Printf(": %s\n", data)
+	}
+}
```

Comment:
the `else` in almost all of these is unnecessary if you just return from the first `if`. For example

```go
func (data ConfigString) ToString() {
  if data == "" {
    fmt.Println(": not set")
    return
  }
  fmt.Printf(": %s\n", data)
}

```